vendor/bat-native-ads/src/bat/ads/internal/ml/data/vector_data_unittest.cc

```diff
@@ -40,6 +40,17 @@ TEST_F(BatAdsVectorDataTest, SparseVectorDataInitialization) {
   EXPECT_EQ(kDimensionCount, sparse_data_vector_6.GetDimensionCount());
 }
 
+TEST_F(BatAdsVectorDataTest, StringDataInitialization) {
+  // Arrange
+  const std::string vector_string = "0.1 0.2 0.3 0.4 0.5 0.6";
+  const VectorData string_data_vector_6(vector_string);
+
+  // Act
+
+  // Assert
+  EXPECT_EQ(6, string_data_vector_6.GetDimensionCount());
```

Comment:
This won't check if the values themselves are correct. I would add a line to initialize a VectorData object using the vector (non-string) way and then check if both initializations match each other